WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -409,6 +447,56 @@ class MediaLibraryViewController: UIViewController {
         }
     }
 
+    // MARK: - Document Picker
+    
+    private func showDocumentPicker() {
+        let docTypes = [String(kUTTypeImage), String(kUTTypeAudiovisualContent)]
+        let docPicker = UIDocumentPickerViewController(documentTypes: docTypes, in: .import)
+        docPicker.delegate = self
+        self.present(docPicker, animated: true, completion: nil)
+    }
+    
+    func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentsAt urls: [URL]) {
+        showPreparingProgressHUD()
+        mediaProgressCoordinator.track(numberOfItems: urls.count)
+        // Wait until all assets are uploaded before we update the collection view
+        pickerDataSource.isPaused = true
+        
+        for documentURL in urls {
+            let mediaID = documentURL.lastPathComponent
+            let service = MediaService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+            service.createMedia(url: documentURL,
+                                forPost: self.blog.objectID,
```

Comment:
You're passing a `blog`'s object ID here, instead of a `post`'s one. You might need to duplicate one of the methods in `MediaService` to take a URL and a `blogObjectID` (the media library VC doesn't have access to a single post to add media to, only a blog).